end-to-end/walkthrough.md

```diff
@@ -0,0 +1,769 @@
+# Attestation Walkthough
+
+This guide explains how to setup a verifier and an attester. To produce a workable example setup, it is necessary to 
+
+- create attestation keys on the attester,
+- provisioning trust anchors, reference values and endorsements on the verifier, and
+- to create evidence on the attester for consumption by the verifier.
+
+For this example setup evidence is provided in form of an PSA attestation token.
+
+## Software Dependencies
+
+The software depends at least on 
+
+- Docker X, and
+- Go
+
+Veraison also has additional dependencies, which are described at https://github.com/veraison/services
+
+Important: Do not install Docker Desktop (at least on Ubuntu) since otherwise Veraison will fail.
+
+Use the following command to install the dependencies for Veraison on an Ubuntu OS:
+
+~~~
+sudo apt install git docker.io jq tmux docker-buildx
+~~~
+
+## Veraison
+
+This section provides details about the use of the Veraison docker image, which is meant to be used for demo purporses. The repository of Veraison can be found at https://github.com/veraison/services
+
+In summary, the following steps need to be executed:
+
+~~~
+git clone https://github.com/veraison/services.git
+cd services
+make docker-deploy
+
+source deployments/docker/env.bash
+veraison status
+~~~
+
+If Veraison services are not running, then they need to be started with the following command:
+
+~~~
+veraison start
+~~~
+
+To check the status of the database use the following command:
+
+~~~
+veraison check-stores
+~~~
+
+Of course, after the initial installation the database will be empty. Provisioning of reference values, trust anchors and endosements will
+be explained below and those will be stored in the database. If you haven't created attestation keys yet, then first look at the next section otherwise skip it. 
+
+In case of problems, enable debugging by issuing the following command in a separate command line window. 
```

Comment:
"Enable debugging" is probably not the right term here. The command just follows STDOUT from the service. The level of output is not changed, and the logs can also be obtained after the fact with `veraison logs`.

Debug output can be enabled by changing the log level inside `deployments/docker/src/config.yaml.template` prior to creating the deployment (i.e. running `make docker-deploy`; if a deployment already exists `make really-clean` should be done first).

(See also https://github.com/veraison/services/tree/main/deployments/docker#debugging for more detailed instruction on how to debug a specific service).